src/__test__/authorize.test.ts

```diff
@@ -101,3 +101,77 @@ describe('authorize - with secret as callback in options', () => {
     })
   })
 })
+
+describe('authorize - with onAuthentication callback in options', () => {
+  let token: string = ''
+  let tokenWrong: string = ''
+
+  beforeEach(async (done) => {
+    jest.setTimeout(15_000)
+    await fixtureStart(
+      async () => {
+        const response = await axios.post('http://localhost:9000/login')
+        token = response.data.token
+        const responseWrong = await axios.post('http://localhost:9000/login-wrong')
+        tokenWrong = responseWrong.data.token
+        done()
+      },
+      {
+        secret: secretCallback,
+        onAuthentication: decodedToken => {
+          if (decodedToken.checkField === true) {
+            return {
+              email: decodedToken.email
+            }
+          } else {
```

Comment:
When, we are in a function, I'll avoid to use too much `else` block, since we have the `return` keyword.
What we could do instead is to do first the negative check, like that : `if (!decodedToken.checkField)` and throw the error and at the end, return the email without the else block.